core/src/main/java/com/sequenceiq/cloudbreak/service/publicendpoint/GatewayPublicEndpointManagementService.java

```diff
@@ -172,7 +178,12 @@ public void deleteLoadBalancerDnsEntry(Stack stack, String environmentName) {
             environmentName = environment.getName();
         }
 
-        for (LoadBalancer loadBalancer : loadBalancers) {
+        Optional<LoadBalancer> loadBalancerOptional = selectLoadBalancer(loadBalancers);
```

Comment:
this logic seems to be duplicated. I think it would be beneficial the refactor it